src/interfaces/coral_web/src/hooks/snapshots.ts

```diff
@@ -0,0 +1,110 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+
+import {
+  GetSnapshotV1SnapshotsLinkLinkIdGetResponse,
+  ListSnapshotsV1SnapshotsGetResponse,
+  Snapshot,
+  SnapshotData,
+  useCohereClient,
+} from '@/cohere-client';
+import { ChatMessage } from '@/types/message';
+
+type FormattedSnapshotData = Omit<SnapshotData, 'messages'> & { messages?: ChatMessage[] };
+export type ChatSnapshot = Omit<Snapshot, 'snapshot'> & { snapshot: FormattedSnapshotData };
+export type ChatSnapshotWithLinks = ChatSnapshot & { links: string[] };
+
+/**
+ * @description returns a list of snapshots
+ */
+const useListSnapshots = () => {
+  const client = useCohereClient();
+
+  return useQuery<ListSnapshotsV1SnapshotsGetResponse, Error>({
+    queryKey: ['listSnapshots'],
+    queryFn: async () => {
+      try {
+        return await client.listSnapshots();
+      } catch (e) {
+        throw e;
```

Comment:
to be consistent with the rest of this file:
```suggestion
        console.error(e);
        throw e;
```